### PR TITLE
Add functionality to TrainSpeedFault

### LINKED ISSUE #316: Add functionality to `TrainSpeedFault`
## User stories
As a developer I want functionality in `TrainSpeedFault` in order to manipulate the speed of trains in the simulation

## Acceptance Criteria
- [ ] implement `inject_fault`
- [ ] implement `resolve_fault`
- [ ] write tests

### PR DESCRIPTION
**Pull request by instant-sky**
Fixes #316 

Description (what might a PO or reviewer want to know?)
Adds functionality to the TrainSpeedFault

## PR checklist

- [ ] Acceptance criteria fulfilled
- [ ] Additional features are tested
- [ ] Docs (code, wiki & diagrams) updated
- [ ] Breaking changes anounced
- [ ] Added classes that inherit from `BaseModel` to `src.constants.tables`
- [ ] Dev-branch has been merged into local branch to resolve conflicts
- [ ] Tests and linter have passed AFTER local merge
- [ ] Another dev reviewed and approved

### CODE REVIEW

File: src/fault_injector/fault_types/train_speed_fault.py [hunk 1]
```diff
@@ -1,27 +1,68 @@
...
-        # - save the current speed of the train in old_speed
-        # - set train speed to new_speed
-        raise NotImplementedError()
+        self.train: Train = [
+            train
+            for train in self.wrapper.trains
+            if train.identifier == self.configuration.affected_element_id
+        ][0]
+        self.old_speed = self.train.train_type.max_speed
+        self.train.train_type.max_speed = self.configuration.new_speed
```
Comment (CR1337) [inline src/fault_injector/fault_types/train_speed_fault.py:49]: Doesn't this change the `may_speed` for every train of this type?

File: tests/fault_injector/test_train_speed_fault.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        train: Train,
+        # the following arguments are needed fixtures
+        # pylint: disable=unused-argument
+        max_speed,
+        train_add,
+        combine_train_and_wrapper,
+        # pylint: enable=unused-argument
+    ):
+        train.train_type.max_speed = 100
+        assert train.train_type.max_speed == 100
```
Comment (CR1337) [inline tests/fault_injector/test_train_speed_fault.py]: Is it possible, that this is not the case?

**Review by CR1337 — COMMENTED** [review]
Looks good :+1: 

I have just two small questions :)

File: src/fault_injector/fault_types/train_speed_fault.py (hunk 1 above)
Comment (instant-sky) [reply src/fault_injector/fault_types/train_speed_fault.py:49]: I thought so too, but @SaturnHafen said it only changes `max_speed` of this specific train...

File: tests/fault_injector/test_train_speed_fault.py (hunk 2 above)
Comment (instant-sky) [reply tests/fault_injector/test_train_speed_fault.py]: I don't know, do you have a case in mind where this does not apply?

**Review by CR1337 — APPROVED** [review]
:+1:

**Pull request merged**